### PR TITLE
Adding E2E tests for EventGrid output binding

### LINKED ISSUE #723: Add E2E Tests for EventGrid Output Binding
Add tests for the changes done in https://github.com/Azure/azure-functions-python-library/pull/56

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
        
This test needs three functions to work.
1. `eventgrid_output_binding`
2. `eventgrid_output_binding_message_to_blobstore`
3. `eventgrid_output_binding_success`

This test calls the eventgrid_output_binding function, sends in a unique UUID as `data` in the body to the httpTrigger which sends in that value in the eventGrid output data. The eventGrid topic is configured to send the event to a storage queue.

The second function (`eventgrid_output_binding_message_to_blobstore`) reads from that storage queue and puts into a blob store.

The third function (`eventgrid_output_binding_success`) reads the text from the blob store and compares with the expected result. The unique uuid should confirm if the message went through correctly to EventGrid and came back as a blob.

Fixes #723

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/endtoend/test_eventgrid_functions.py [hunk 1]
```diff
@@ -69,17 +86,82 @@ def test_eventgrid_trigger(self):
...
+
+        The third function (`eventgrid_output_binding_success`) reads the
+        text from the blob store and compares with the expected result. The
+        unique uuid should confirm if the message went through correctly to
+        EventGrid and came back as a blob.
+        """
+
+        test_uuid = uuid.uuid4().__str__()
+
+        data = "{" + "'test_uuid': '{0}'".format(test_uuid) + "}"
```
Comment (Hazhzeng) [inline tests/endtoend/test_eventgrid_functions.py:125]: Using f"string" would be cleanner in this case (Python >= 3.6)
```python
data = f"{{ 'test_uuid': '{test_uuid}' }}"
```

File: tests/endtoend/test_eventgrid_functions.py [hunk 2]
```diff
@@ -69,17 +86,82 @@ def test_eventgrid_trigger(self):
...
+        expected_response = "Sent event with subject: {}, id: {}, data: {}, " \
+                            "event_type: {} to EventGrid!".format(
+                                "test-subject", "test-id", data, "test-event-1")
+        expected_final_data = {
+            'id': 'test-id', 'subject': 'test-subject', 'dataVersion': '1.0',
+            'eventType': 'test-event-1',
+            'data': {'test_uuid': test_uuid}
+        }
```
Comment (Hazhzeng) [inline tests/endtoend/test_eventgrid_functions.py:133]: I think this is pretty cool to place expected result at the top of a test case, I will follow this flow when writing test cases.

File: azure_functions_worker/testutils.py [hunk 3]
```diff
@@ -640,6 +640,15 @@ def popen_webhost(*, stdout, stderr, script_root=FUNCS_PATH, port=None):
         if servicebus:
             extra_env['AzureWebJobsServiceBusConnectionString'] = servicebus
 
+        eventgrid_topic_uri = testconfig['azure'].get('eventgrid_topic_uri')
```
Comment (Hazhzeng) [inline azure_functions_worker/testutils.py:643]: May want to update the wiki->contributor guide, step 7 for this newly introduced config **eventgrid_topic_uri** and **eventgrid_topic_key**

File: tests/endtoend/test_eventgrid_functions.py [hunk 4]
```diff
@@ -69,17 +86,82 @@ def test_eventgrid_trigger(self):
...
                 self.assertEqual(
-                    response,
-                    {
-                        'id': data[0]['id'],
-                        'data': data[0]['data'],
-                        'topic': data[0]['topic'],
-                        'subject': data[0]['subject'],
-                        'event_type': data[0]['eventType'],
-                    }
+                    response, {'id': data[0]['id'], 'data': data[0]['data'],
```
Comment (Hazhzeng) [inline tests/endtoend/test_eventgrid_functions.py:89]: Syntactic sugar when comparing two dicts, checking if response is a subset of data[0]:
```python
self.assertLessEqual(response.items(), data[0].items())
```

File: tests/endtoend/test_eventgrid_functions.py (hunk 1 above)
Comment (vrdmr) [reply tests/endtoend/test_eventgrid_functions.py:125]: Thanks. Yes, I tried that earlier inside the `format()` in the next line, but that was throwing some other issue. Fixed it.

File: tests/endtoend/test_eventgrid_functions.py (hunk 2 above)
Comment (vrdmr) [reply tests/endtoend/test_eventgrid_functions.py:133]: It depends :)
In some cases having expected value closer to the assert makes sense.

File: azure_functions_worker/testutils.py (hunk 3 above)
Comment (vrdmr) [reply azure_functions_worker/testutils.py:643]: Sure. I'll do that post-merge.

**Pull request merged**